rust/cubesqlplanner/cubesqlplanner/src/physical_plan_builder/builder.rs

```diff
@@ -0,0 +1,1138 @@
+            return Err(CubeError::internal(format!(
+                "Schema not found for cte {}",
+                measure_input_ref
+            )));
+        };
+
+        let base_time_dimension_alias = measure_input_schema.resolve_member_alias(
+            &rolling_window
+                .time_dimension_in_measure_input
+                .clone()
+                .as_base_member(self.query_tools.clone())?,
+        );
+
+        let root_alias = format!("time_series");
+        let measure_input_alias = format!("rolling_source");
+
+        let mut join_builder = JoinBuilder::new_from_table_reference(
+            time_series_ref.clone(),
+            time_series_schema,
+            Some(root_alias.clone()),
+        );
+
+        let on = match &rolling_window.rolling_window {
+            MultiStageRollingWindowType::Regular(regular_rolling_window) => {
+                JoinCondition::new_regular_rolling_join(
+                    root_alias.clone(),
+                    regular_rolling_window.trailing.clone(),
+                    regular_rolling_window.leading.clone(),
+                    regular_rolling_window.offset.clone(),
+                    Expr::Reference(QualifiedColumnName::new(
+                        Some(measure_input_alias.clone()),
+                        base_time_dimension_alias,
+                    )),
+                )
+            }
+            MultiStageRollingWindowType::ToDate(to_date_rolling_window) => {
+                JoinCondition::new_to_date_rolling_join(
+                    root_alias.clone(),
+                    to_date_rolling_window.granularity.clone(),
+                    Expr::Reference(QualifiedColumnName::new(
+                        Some(measure_input_alias.clone()),
+                        base_time_dimension_alias,
+                    )),
+                    self.query_tools.clone(),
+                )
+            }
+            MultiStageRollingWindowType::RunningTotal => JoinCondition::new_rolling_total_join(
+                root_alias.clone(),
+                Expr::Reference(QualifiedColumnName::new(
+                    Some(measure_input_alias.clone()),
+                    base_time_dimension_alias,
+                )),
+            ),
+        };
+
+        join_builder.left_join_table_reference(
+            measure_input_ref.clone(),
+            measure_input_schema.clone(),
+            Some(measure_input_alias.clone()),
+            on,
+        );
+
+        let mut context_factory = context.make_sql_nodes_factory();
+        context_factory.set_rolling_window(true);
+        let from = From::new_from_join(join_builder.build());
+        let references_builder = ReferencesBuilder::new(from.clone());
+        let mut render_references = HashMap::new();
+        let mut select_builder = SelectBuilder::new(from.clone());
+
+        //We insert render reference for main time dimension (with the some granularity as in time series to avoid unnessesary date_tranc)
```

Comment:
```suggestion
        //We insert rendered reference for the main time dimension (with the same granularity as in time series to avoid unnecessary date_tranc)
```